t/12_sles4sap_publiccloud.t

```diff
@@ -535,6 +537,50 @@ subtest '[get_hana_topology]' => sub {
     ok((any { qr/SAPHanaSR-showAttr --format=/ } @calls), 'function calls SAPHanaSR-showAttr');
 };
 
+subtest '[get_hana_topology_json]' => sub {
+    my @calls;
+    my $self = sles4sap_publiccloud->new();
+    set_var('USE_SAP_HANA_SR_ANGI', 'true');
+    my $sles4sap_publiccloud = Test::MockModule->new('sles4sap_publiccloud', no_auto => 1);
+    $sles4sap_publiccloud->redefine(record_info => sub { note(join(' ', 'RECORD_INFO -->', @_)); });
+    my %test_topology = (
+        Host => {
+            vmhanaBBBBB => {
+                vhost => 'vmhanaBBBBB',
+                site => 'site_b'
+            },
+            vmhanaAAAAA => {
+                site => 'site_a',
+                vhost => 'vmhanaAAAAA',
+            }
+        },
+        Site => {
+            site_b => {
+                lss => '4',
+                mns => 'vmhanaAAAAA',
+                srPoll => 'SOK',
+            },
+            site_a => {
+                lss => '4',
+                mns => 'vmhanaBBBBB',
+                srPoll => 'PRIM',
+            }
+        }
+    );
+    $sles4sap_publiccloud->redefine(calculate_hana_topology => sub { return encode_json \%test_topology; });
+    $sles4sap_publiccloud->redefine(run_cmd => sub {
+            my ($self, %args) = @_;
+            push @calls, $args{cmd};
+            return "Output does no matter as calculate_hana_topology is redefined.";
+    });
+    $sles4sap_publiccloud->redefine(wait_for_idle => sub { return; });
+    my $topology = decode_json $self->get_hana_topology();
+    note("\n  C -->  " . join("\n  C -->  ", @calls));
+    ok((keys %{$topology->{Host}} eq 2 && %{$topology->{Site}} eq 2), 'Two hosts and two sites returned by calculate_hana_topology');
```

Comment:
considering what you decide to `redefine`, line 579 and 581 in the test are not testing anything from the implementation of the function under test.

I think only line 582 is ok to be here: it is testing the `get_hana_topology` command line composition: so these two lines https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/21545/files#diff-4b7291e584e1d0b487f685a7b45efc74dcd804ee2236acf7a866003d0f4fb87aR249